components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java

```diff
@@ -172,4 +213,174 @@ public void testIdentityOAuth2Exception() throws Exception {
+        mockStatic(AccessTokenIssuer.class);
+        when(AccessTokenIssuer.getInstance()).thenReturn(accessTokenIssuer);
+        when(accessTokenIssuer.issue(any(OAuth2AccessTokenReqDTO.class))).thenReturn(tokenRespDTO);
+        assertNotNull(oAuth2Service.issueAccessToken(new OAuth2AccessTokenReqDTO()));
+
+        when(accessTokenIssuer.issue(any(OAuth2AccessTokenReqDTO.class)))
+                .thenThrow(new InvalidOAuthClientException(""));
+        assertEquals(oAuth2Service.issueAccessToken(new OAuth2AccessTokenReqDTO())
+                .getErrorMsg(), "Invalid Client");
+    }
+
+
+    @Test
+    public void testExceptionForIssueAccesstoken() throws IdentityException {
+
+        AccessTokenIssuer accessTokenIssuer = mock(AccessTokenIssuer.class);
+        mockStatic(AccessTokenIssuer.class);
+        when(AccessTokenIssuer.getInstance()).thenReturn(accessTokenIssuer);
+        when(accessTokenIssuer.issue(any(OAuth2AccessTokenReqDTO.class)))
+                .thenThrow(new IdentityOAuth2Exception(""));
+        assertEquals(oAuth2Service.issueAccessToken(new OAuth2AccessTokenReqDTO())
+                .getErrorCode(), "server_error");
+    }
+
+    @Test
+    public void testIsPKCESupportEnabled() {
+
+        OAuthServerConfiguration oAuthServerConfiguration = mock(OAuthServerConfiguration.class);
+        when(oAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn(3600L);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        mockStatic(OAuth2Util.class);
+        when(OAuth2Util.isPKCESupportEnabled()).thenReturn(true);
+        assertTrue(oAuth2Service.isPKCESupportEnabled());
+    }
+
+    @Test
+    public void testRevokeTokenByOAuthClientWithRefreshToken() throws Exception {
+
+        setUpRevokeToken();
+
+        RefreshTokenValidationDataDO refreshTokenValidationDataDO = new RefreshTokenValidationDataDO();
+        refreshTokenValidationDataDO.setGrantType(GrantType.REFRESH_TOKEN.toString());
+        refreshTokenValidationDataDO.setAccessToken("testAccessToken");
+        refreshTokenValidationDataDO.setAuthorizedUser(authenticatedUser);
+        refreshTokenValidationDataDO.setScope(new String[]{"test"});
+        refreshTokenValidationDataDO.setRefreshTokenState(OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE);
+
+        TokenMgtDAO tokenMgtDAO = mock(TokenMgtDAO.class);    //mock methods for tokenMgtDAO
+        when(tokenMgtDAO.validateRefreshToken(anyString(), anyString())).thenReturn(refreshTokenValidationDataDO);
+        whenNew(TokenMgtDAO.class).withAnyArguments().thenReturn(tokenMgtDAO);
+        doNothing().when(tokenMgtDAO).revokeTokens(any(String[].class));
+
+        OAuthRevocationRequestDTO revokeRequestDTO = new OAuthRevocationRequestDTO();
+        assertEquals(oAuth2Service.revokeTokenByOAuthClient(revokeRequestDTO).getErrorMsg(),
```

Comment:
Here when running multiple tests within a single test method, if the very first assert fails none of the other logic will execute. Better to use a data provider and write this as a single test.